### PR TITLE
Fix distance waypoint ordering

### PR DESCRIPTION
**Pull request by neil-marcellini**
### Details

The changes here are pretty simple and are fixing things that worked before. First, when there were 10+ waypoints they would get out of order within `getValidWaypoints` because we were sorting them as strings. That regression is described [here](https://github.com/Expensify/App/pull/28005#discussion_r1362951809). Second, the parameters of a function were changed without updating this use. That regression is noted [here](https://github.com/Expensify/App/commit/aa07b9c77dd0e2a317327dd0502a51abc1cc2faf#r130356775).

### Fixed Issues

$ https://github.com/Expensify/App/issues/29621
PROPOSAL: N/A

### Tests

Test waypoint ordering
1. Log in with any account
2. create a workspace if needed
6. Click the green plus and Request money
7. Select the distance tab
8. Enter 12 waypoints total (or use the shortcut below), and make note of the order
9. Click next
10. Select a workspace
11. Click request
12. Open the expense report
13. Open the distance request report
14. Click on the route image
15. Verify the distance request eReceipt is displayed
16. Verify the route image is displayed
17. Verify that the waypoints are in the same order as when creating the request
---

**Note:** since it's a pain to enter all those waypoints here's a shortcut.
1. Open the console and switch to the distance tab
3. Make sure all log levels are shown
4. Seach for `transactions_` and copy the id number
5. Paste the following into the JS console to set all the waypoints. Replace the id with the one you copied.
<details>
<summary>Onyx update</summary>

```
Onyx.merge('transactions_3361518241221798589', {
    comment: {
        waypoints: {
            "waypoint0": {
                "lat": 37.9004816,
                "lng": -122.6444263,
                "address": "Stinson Beach, CA 94970, USA"
            },
            "waypoint1": {
                "lat": 37.8652634,
                "lng": -122.3111483,
                "address": "Berkeley Marina, Berkeley, CA, USA"
            },
            "waypoint2": {
                "lat": 37.8064291,
                "lng": -122.4506915,
                "address": "Crissy Field East Beach, San Francisco, CA 94129, USA"
            },
            "waypoint3": {
                "lat": 37.8727428,
                "lng": -122.4554066,
                "address": "Tiburon, CA 94920, USA"
            },
            "waypoint4": {
                "lat": 37.9445266,
                "lng": -122.5086664,
                "address": "Larkspur, CA 94939, USA"
            },
            "waypoint5": {
                "lat": 37.8974259,
                "lng": -122.4885845,
                "address": "Blackies Pasture, Tiburon, CA 94920, USA"
            },
            "waypoint6": {
                "lat": 36.6002378,
                "lng": -121.8946761,
                "address": "Monterey, CA, USA"
            },
            "waypoint7": {
                "lat": 37.8309974,
                "lng": -122.2919998,
                "address": "4400 Shellmound St, Emeryville, CA 94608, USA"
            },
            "waypoint8": {
                "lat": 37.9435872,
                "lng": -122.5405003,
                "address": "800 Magnolia Ave, Larkspur, CA 94939, USA"
            },
            "waypoint9": {
                "lat": 37.6191671,
                "lng": -122.3816108,
                "address": "San Francisco, CA 94128, USA"
            },
            "waypoint10": {
                "lat": 37.7156188,
                "lng": -122.2143079,
                "address": "1 Airport Dr, Oakland, CA 94621, USA"
            },
            "waypoint11": {
                "lat": 37.7935724,
                "lng": -122.483638,
                "address": "Baker Beach, San Francisco, CA, USA"
            }
        }
    }
});
```
</details>

- [ ] Verify that no errors appear in the JS console

### Offline tests

1. Go offline
2. Create the request as described above 
3. Open the eReceipt
4. Verify that route image shows as pending
5. Go online
6. Verify that the waypoints are in the same order as when creating the request

### QA Steps

Same as tests
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
I only tested on Mac web because the changes are very small and platform independent.

<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Online

https://github.com/Expensify/App/assets/26260477/ea2144c2-f6a9-447b-8506-2dc50d569a1d

Offline

https://github.com/Expensify/App/assets/26260477/d58fc2c8-2f58-4c34-9ba4-7078f212d718

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Review by grgia — APPROVED** [review]
LGTM

File: src/components/DistanceEReceipt.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ const defaultProps = {
 function DistanceEReceipt({transaction}) {
     const {translate} = useLocalize();
     const {isOffline} = useNetwork();
-    const {thumbnail} = TransactionUtils.hasReceipt(transaction) ? ReceiptUtils.getThumbnailAndImageURIs(transaction.receipt.source, transaction.filename) : {};
+    const {thumbnail} = TransactionUtils.hasReceipt(transaction) ? ReceiptUtils.getThumbnailAndImageURIs(transaction) : {};
```
[the remark this answers is not in the document]
Comment (neil-marcellini) [reply src/components/DistanceEReceipt.js:34]: Ok, but I don't think that prevents you from reviewing right? Will you please review today?

[the remark this answers is not in the document]
Comment (situchan) [reply src/components/DistanceEReceipt.js:34]: right

**Comment by situchan** [comment]
Not sure what's happening but I am not able to submit distance request on latest main.
Request button is disabled

https://github.com/Expensify/App/assets/108292595/52fda258-4cdf-4dd5-9f21-676e3c356dab

**Comment by situchan** [comment]
I confirmed that https://github.com/Expensify/App/pull/27547 caused above regression. [Commented there](https://github.com/Expensify/App/pull/27547#issuecomment-1771692897)

**Comment by neil-marcellini** [comment]
Thanks for helping me push this one through guys!

**Pull request merged**